projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+    "bufio"
+    "fmt"
+    "net/http"
+	"os"
+	"time"
+	"strconv"
+	"errors"
+	"io"
+)
+
+func main() {
+    resp, err := http.Get("http://localhost:8080")
+
+   if err != nil {
+		if err != nil {
+		if errors.Is(err, io.EOF) {
+			fmt.Println("The server closed the connection unexpectedly. Please try again later.")
+		} else {
+			fmt.Println("Error:", err)
+		}
+		os.Exit(1)
+	}
+	}
+    defer resp.Body.Close()
+
+    fmt.Println("Response status:", resp.Status)
+
+    scanner := bufio.NewScanner(resp.Body)
+    for i := 0; scanner.Scan() && i < 10; i++ {
+        fmt.Println(scanner.Text())
+    }
+
+    if err := scanner.Err(); err != nil {
+        panic(err)
+    }
+	if resp.StatusCode == http.StatusTooManyRequests {
+        handleTooManyRequests(resp)
+    }
+}
+
+func handleTooManyRequests(resp *http.Response) {
+    retryAfterHeader := resp.Header.Get("Retry-After")
+    if retryAfterHeader != "" {
```

Comment:
This works, but a common pattern is to check this the other way around and early-return if it's not what we expect, e.g.

```go
if retryAfterHeader == "" {
    // Maybe log something here
    return
}
```

Have a read of https://medium.com/swlh/return-early-pattern-3d18a41bba8 to see some of the reasons why this is beneficial